addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -282,16 +314,17 @@ def _l10n_eg_eta_prepare_invoice_lines_data(self, invoice, tax_data):
                 },
                 'taxableItems': [
                     {
-                        'taxType': tax['tax_repartition_line'].tax_id.l10n_eg_eta_code.split('_')[0].upper().upper(),
-                        'amount': self._l10n_eg_edi_round(abs(tax['tax_amount'])),
-                        'subType': tax['tax_repartition_line'].tax_id.l10n_eg_eta_code.split('_')[1].upper(),
-                        **({'rate': abs(tax['tax_repartition_line'].tax_id.amount)} if tax['tax_repartition_line'].tax_id.amount_type != 'fixed' else {}),
+                        'taxType': grouping_key['tax_type'],
+                        'amount': self._l10n_eg_edi_round(tax_values['tax_amount']),
+                        'subType': grouping_key['sub_type'],
+                        'rate': grouping_key['rate'],
```

Comment:
so if tax.amount_type == 'fixed', grouping_key['rate'] is falsy?

Reply:
I'm handling this when creating the grouping key here:
https://github.com/odoo/odoo/pull/180062/files#diff-9f344af59ad383237995c468948772716770a6d092aafc98008bc0e0d9a35dfbR230